### ISSUE TITLE
Bad output results in production inference; model trained using Trainer(precision=16)

### ISSUE THREAD

**Issue by albertmundu** [PyTorchLightning/pytorch-lightning#12849]
Hello PL team,
I am getting bad/random noise results/outputs from the model in inference when the model is trained using `Trainer(precision=16)`. I have also tested using `trainer.test` and it is producing good results for it. Only for inference it is not working fine.

My sample code for inference
```
 with autocast():
        model = ModelTrainer.load_from_checkpoint("best_checkpoints/last.ckpt").to(device)
        model.eval()
        with torch.no_grad():
            x = x.to(device)
            predicted_img  = model(x)

```
How can I solve this issue?

**Comment by albertmundu** [comment]
Solved the issue. The problem is not about precision. It about saving hyperparameters. I used the learning rate directly to the optimizer (hard coded) and had not initialized the learning rate in the `__init__` method. Adding it, solved my problem. 
```
class ModelTrainer(pl.LightningModule)
def __init__():
        self.save_hyperparameters()
        self.model = ModelTrainer(in_channels=2, out_channels=1)
        # self.ssim = StructuralSimilarityIndexMeasure()
        self.rmse = MeanSquaredError(squared=False)
        self.learning_rate = 1e-3 # added to solve the problem

def configure_optimizers(self):
        # removed the hard coded value with lr=self.learning_rate
        optimizer = torch.optim.Adam(self.parameters(), lr=self.learning_rate)
        return optimizer
```
